src/chapters/setup-intro-data-types/more-data-types.rst

```diff
@@ -2,258 +2,321 @@ More Data Types
+terminology, ``myCat`` is a *reference variable* of type ``Cat``.
 
 References
 ^^^^^^^^^^^
 
-Reference types are different from primitive types in an essential way.
-A variable of a reference type (such as ``myCat`` above) does not
-actually store the object in question. Instead, it stores a
-**reference** to the object. A reference is literally a memory address.
-We visualize references as an arrow pointing to the object in memory.
+Reference types are different from primitive types in an essential way. A
+variable of a reference type (such as ``myCat`` above) does not actually store
+the object in question. Instead, it stores a *reference* to the object, which
+is literally a memory address. We visualize references as an arrow pointing to
+the location of the object in memory.
 
 Consider this code:
 
-.. code:: java
+.. sourcecode:: java
+   :linenos:
 
-   int catAge = 11;
+   int firstCatAge = 11;
+   int secondCatAge = firstCatAge;
    Cat myCat = new Cat();
    Cat sameCat = myCat;
 
-Visually, we can represent these three variables as shown below.
+Visually, we can represent these four variables as shown below.
 
 .. figure:: figures/references.png
    :alt: Reference Types
 
-   Reference Types
+   Reference Types vs. Primitive Types
 
-Since ``int`` is a primitive type, the variable ``catAge`` functions as
-a box holding the integer value 11. On the other hand, ``myCat`` is a
-reference variable, since it holds an object. The variable actually
-stores the address of the object, which we visualize as an arrow from
-the variable box to the object.
+Since ``int`` is a primitive type, the variables ``firstCatAge`` and
+``secondCatAge`` function like separate boxes, each one holding the integer
+value ``11``. On the other hand, ``myCat`` is a reference variable, since it
+refers to an object of type ``Cat``. The variable actually stores the *memory
+address* of the object, which we visualize as an arrow pointing from the
+variable box to where the data is stored. Instead of holding the actual ``Cat``
+data, ``myCat`` stores *directions* for finding the data in memory.
 
-When we assign a value to a reference type, as in
-``Cat sameCat = myCat``, we are not creating a second copy of the
-object, but instead are creating a second “arrow” pointing to the same
-object.
+When we assign a value to a reference type, as in ``Cat sameCat = myCat``, we
+do NOT create a second copy of the object or its data. Instead, we make a
+second arrow pointing to the same memory location.
 
-The distinction between references types and primitives is important, if
-difficult, to wrap your brain around at first. We will see that
-reference types are handled differently in essential and important ways
-in a lot of different situations.
+The distinction between reference types and primitives is important, if
+subtle. As you continue learning Java, you will see that reference types are
+handled differently in essential and important ways.
```

Comment:
good section! i feel like i may not be the only one to read this and then wonder about the relationship between object types and reference types?